sympy/functions/special/bessel.py

```diff
@@ -375,11 +375,15 @@ def _eval_nseries(self, x, n, logx, cdir=0):
                 return o
             t = (_mexpand(r**2) + o).removeO()
 
-            if nu > S.One:
+            if nu > S.Zero:
```

Comment:
What if `nu` is zero?

Reply:
Nothing much actually. The series is given by 
<img width="634" alt="image" src="https://user-images.githubusercontent.com/87052487/182006733-6f64a107-a38f-40c5-b96d-5dc12b14ed0e.png">
And `nu` being zero only means that we won't be able to get any terms out of the second summation cause the upper limit (nu - 1 falls to -1 then) and other things remain the same. So we get our answers as we would expect them to be
```
>>> bessely(0, x).series(x, 0, 3)
2*EulerGamma/pi - 2*log(2)/pi + 2*log(x)/pi + x**2*(-log(x)/(2*pi) + (2 - 2*EulerGamma)/(4*pi) + log(2)/(2*pi)) + O(x**3*log(x))
>>> bessely(0, sin(x)).series(x, 0, 3)
2*EulerGamma/pi - 2*log(2)/pi + 2*log(x)/pi + x**2*(-log(x)/(2*pi) - 1/(3*pi) + (2 - 2*EulerGamma)/(4*pi) + log(2)/(2*pi)) + O(x**3*log(x))
```

Reply:
So could we have `nu >= 0` here?

Reply:
No right , when `nu = 0`, we would have something like
```
>>> Sum(f(x), (x, 0, -1)).doit()
```
And like we don't want this , when `nu = 0` we don't expect it to enter the for loop which calculates the second summation, it should just skip that block. Otherwise we land up with something like
```
>>> bessely(0, x).series(x, 0, 3)
zoo + O(x**3*log(x))
```

Reply:
> it should just skip that block. 

Would we get the correct result for `nu == 0` if the sum block is skipped?

Reply:
Yes !